homeassistant/components/teslemetry/select.py

```diff
@@ -0,0 +1,259 @@
+    SeatHeaterDescription(
+        key="climate_state_seat_heater_third_row_right",
+        position=Seat.THIRD_RIGHT,
+        available_fn=lambda self: self.get("vehicle_config_third_row_seats") != "None",
+        entity_registry_enabled_default=False,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Teslemetry select platform from a config entry."""
+
+    scoped = Scope.VEHICLE_CMDS in entry.runtime_data.scopes
+
+    async_add_entities(
+        chain(
+            (
+                TeslemetrySeatHeaterSelectEntity(vehicle, description, scoped)
+                for description in SEAT_HEATER_DESCRIPTIONS
+                for vehicle in entry.runtime_data.vehicles
+            ),
+            (
+                TeslemetryWheelHeaterSelectEntity(vehicle, scoped)
+                for vehicle in entry.runtime_data.vehicles
+            ),
+            (
+                TeslemetryOperationSelectEntity(energysite, entry.runtime_data.scopes)
+                for energysite in entry.runtime_data.energysites
+                if energysite.info_coordinator.data.get("components_battery")
+            ),
+            (
+                TeslemetryExportRuleSelectEntity(energysite, entry.runtime_data.scopes)
+                for energysite in entry.runtime_data.energysites
+                if energysite.info_coordinator.data.get("components_battery")
+                and energysite.info_coordinator.data.get("components_solar")
+            ),
+        )
+    )
+
+
+class TeslemetrySeatHeaterSelectEntity(TeslemetryVehicleEntity, SelectEntity):
+    """Select entity for vehicle seat heater."""
+
+    entity_description: SeatHeaterDescription
+
+    _attr_options = [
+        OFF,
+        LOW,
+        MEDIUM,
+        HIGH,
+    ]
+
+    def __init__(
+        self,
+        data: TeslemetryVehicleData,
+        description: SeatHeaterDescription,
+        scoped: bool,
+    ) -> None:
+        """Initialize the vehicle seat select entity."""
+        self.entity_description = description
+        self.scoped = scoped
+        super().__init__(data, description.key)
+
+    def _async_update_attrs(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_available = self.entity_description.available_fn(self)
+        value = self._value
+        if value is None:
+            self._attr_current_option = None
+        else:
+            self._attr_current_option = self._attr_options[value]
+
+    async def async_select_option(self, option: str) -> None:
+        """Change the selected option."""
+        self.raise_for_scope()
+        await self.wake_up_if_asleep()
+        level = self._attr_options.index(option)
+        # AC must be on to turn on seat heater
+        if level and not self.get("climate_state_is_climate_on"):
+            await self.handle_command(self.api.auto_conditioning_start())
```

Comment:
Is this mimicking the behavior of the app (to just turn on the climate) without the user knowing about it?
Otherwise could maybe stop the user from turning it on if climate is not on?

Reply:
Yes this mimicks the behavior of the offical Tesla app. If you turn on a seat heater, it turns on the climate control. if you turn off a seat heater, it leaves the clmate on.